File: submissions/0739-daily-temperatures/solution.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function(arr) {
    const n = arr.length;
    const ans = new Array(n).fill(0);
    const st = [];

    for(let i=0; i<n; i++){
        while(st.length && arr[i] > arr[st[st.length-1]]){
            const prevIndex = st.pop();
            ans[prevIndex] = i-prevIndex;
        }
        st.push(i);
    } 
    return ans;
};

File: submissions/0739-daily-temperatures/solution.java

class Solution {
    public int[] dailyTemperatures(int[] arr) {
        int n = arr.length;
        int[] ans = new int[n];
        Stack<Integer> st = new Stack<>();
        for(int i=0; i<n; i++){
            while(!st.isEmpty() && arr[i] > arr[st.peek()]){
                int prevIndex = st.pop();
				ans[prevIndex] = i-prevIndex;
            }
            
            st.push(i);
        }
        return ans;
    }
}
